File: Arrays/P018.py
def brute(nums):
    n=len(nums)
    for i in range(n):
        c=0
        for j in range(n):
            if nums[i]==nums[j]:
                c+=1
        if c!=2:
            return nums[i]
    return -1

def optimal(nums):
    n=len(nums)
    res=nums[0]
    for i in range(1,n):
        res^=nums[i]
    return res

if  __name__=='__main__':
    nums=input('Enter elements: ').split()
    nums=list(map(int,nums))
    print('Brute: ',brute(nums))
    print('optimal: ',optimal(nums))

File: Striver_C++_Problems/Arrays/P018.cpp
// P018.cpp - Find the single element (others appear twice)

#include <bits/stdc++.h>
using namespace std;

int brute(const vector<int>& nums) {
    int n = nums.size();
    for (int i = 0; i < n; i++) {
        int c = 0;
        for (int j = 0; j < n; j++) {
            if (nums[i] == nums[j]) c++;
        }
        if (c != 2) return nums[i];
    }
    return -1;
}

int optimal(const vector<int>& nums) {
    int n = nums.size();
    int res = nums[0];
    for (int i = 1; i < n; i++) {
        res ^= nums[i];
    }
    return res;
}

int main() {
    cout << "Enter elements: ";
    string line;
    getline(cin, line);
    stringstream ss(line);
    vector<int> nums;
    int x;
    while (ss >> x) nums.push_back(x);

    cout << "Brute: " << brute(nums) << "\n";
    cout << "optimal: " << optimal(nums) << "\n";
    return 0;
}
